### PR TITLE
merge from 2.2.0 + fix up board defs

### PR DESCRIPTION
**Pull request by dhalbert**
Merge bugfixes and additions from 2.2.0 to master.

Difficult merge. Checked a number of merges by hand. Non-Adafruit boards added but I did not try to build them.

Merged in 2.2 double-tap support in .ld files (@tannewt: you might double-check these).

CPX SPI flash support added, but not tested.

### CODE REVIEW

**Comment by dhalbert** [comment]
@tannewt Fixed up things that broke the build. @microbuilder, note that there are a few changes in the nrf/ dir to make things compile, and I implemented bidirectional SPI by implementing one new routine (you can check that -- it was a guess based on the the read and write routines).

**Comment by jerryneedell** [comment]
FYI - I was able to merge this PR into master and build/load the code into my feather52 - Seems to be working as expected.

File: ports/atmel-samd/boards/feather_m0_supersized/conf_access.h [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \asf_license_stop
+ *
+ */
+/*
+ * Support and FAQ: visit <a href="http://www.atmel.com/design-support/">Atmel Support</a>
+ */
```
[the remark this answers is not in the document]
Comment (dhalbert) [reply ports/atmel-samd/boards/feather_m0_supersized/conf_access.h]: cleaned up (+more).

File: ports/atmel-samd/boards/ugame/mpconfigboard.mk [hunk 2]
```diff
@@ -0,0 +1,11 @@
+LD_FILE = boards/samd21x18-bootloader-external-flash-crystalless.ld
+#LD_FILE = boards/samd21x18-bootloader.ld
+USB_VID = 0x239A
+USB_PID = 0x801F
+USB_PRODUCT = "UGame"
```
[the remark this answers is not in the document]
Comment (dhalbert) [reply ports/atmel-samd/boards/ugame/mpconfigboard.mk]: tnx

**Comment by dhalbert** [comment]
ready to re-review

**Comment by microbuilder** [comment]
Hi Dan, sorry just saw the email alert for this. :( Looks good to me though!

**Comment by tannewt** [comment]
@microbuilder Feel free to add comments to PRs after they are merged. The author can always follow up with an additional PR.

**Comment by dhalbert** [comment]
All atmel-samd builds are going into safe mode on ctrl-D after this merge (tnx @jerryneedell). So some issue somewhere.

**Pull request merged**